lib/utils.ts

```diff
@@ -90,6 +90,22 @@ export const Utils = new class {
 		} else {
 			constructor = 'null';
 		}
+
+		// if it has a toString, check that to see if it tells us if there's a base class, else look at the constructor name
+		let baseName = value?.toString ? /\[object (.*)\]/.exec(value.toString())?.[1] : null;
+		if (!baseName) baseName = constructor;
+
+		switch (baseName) {
+		// other constructor cases can be handled here!
```

Comment:
This seems kind of superfluous as far as comments go?

```suggestion
```

You could add this comment to any `switch` statement. It's just how `switch`es work. I don't think old language features like `switch` need to be documented this way.

Reply:
Remind me to add a detailed explanation on what comments are for, in CONTRIBUTING, at some point.

Reply:
Did!

https://github.com/smogon/pokemon-showdown/blob/master/CONTRIBUTING.md#dont-teach-javascript